### PR TITLE
fix: Don't multiplexresolve if not needed

### PR DESCRIPTION
**Pull request by disq**

### CODE REVIEW

File: provider/execution/execution.go [hunk 1]
```diff
@@ -66,7 +66,10 @@ func NewTableExecutor(resourceName string, db Storage, logger hclog.Logger, tabl
 // Resolve is the root function of table executor which starts an execution of a Table resolving it, and it's relations.
 func (e TableExecutor) Resolve(ctx context.Context, meta schema.ClientMeta) (uint64, diag.Diagnostics) {
 	if e.Table.Multiplex != nil {
-		return e.doMultiplexResolve(ctx, meta, nil)
+		clients := e.Table.Multiplex(meta)
+		if len(clients) > 0 {
+			return e.doMultiplexResolve(ctx, clients, nil)
+		}
```
Comment (roneli) [inline provider/execution/execution.go:72]: if clients == 1 no need to multiplex either.

**Review by roneli — CHANGES_REQUESTED** [review]
some changes, would also add a test for it in execution.

File: provider/execution/execution.go (hunk 1 above)
Comment (disq) [reply provider/execution/execution.go:72]: yes but multiplexer might've changed something in the return value?

Comment (roneli) [inline provider/execution/execution.go:72]: nit:
```suggestion
		if clients := e.Table.Multiplex(meta); len(clients) > 0 {
			return e.doMultiplexResolve(ctx, clients, nil)
		}
```

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**